File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/QuoteLoader.java
package ro.danserboi.quotesformindandsoul;

import android.content.Context;

import androidx.annotation.Nullable;
import androidx.loader.content.AsyncTaskLoader;

public class QuoteLoader extends AsyncTaskLoader<String> {
    private final Pair<String, String> mQuery;

    QuoteLoader(Context context, Pair<String, String> query) {
        super(context);
        mQuery = query;
    }

    @Nullable
    @Override
    public String loadInBackground() {
        return NetworkUtils.getQuotes(mQuery);
    }

    @Override
    protected void onStartLoading() {
        super.onStartLoading();
        forceLoad();
    }
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/QuoteLoader.kt
package ro.danserboi.quotesformindandsoul

import android.content.Context
import androidx.loader.content.AsyncTaskLoader

class QuoteLoader internal constructor(context: Context?, private val mQuery: Pair<String?, String?>) : AsyncTaskLoader<String?>(context!!) {
    override fun loadInBackground(): String? {
        return NetworkUtils.getQuotes(mQuery)
    }

    override fun onStartLoading() {
        super.onStartLoading()
        forceLoad()
    }
}